core/src/main/c/windows/files.c

```diff
@@ -55,16 +55,76 @@ JNIEXPORT jint JNICALL Java_io_questdb_std_Files_copy
     return -1;
 }
 
-int set_file_pos(HANDLE fd, jlong offset) {
+
+jboolean set_file_pos(HANDLE fd, jlong offset) {
     if (offset < 0) {
         return 1;
     }
-    long highPos = (long) (offset >> 32);
-    int r = SetFilePointer(fd, (LONG) offset, &highPos, FILE_BEGIN) != INVALID_SET_FILE_POINTER;
+    LONG highPos = (LONG) (offset >> 32);
+    DWORD r = SetFilePointer(fd, (LONG) offset, &highPos, FILE_BEGIN) != INVALID_SET_FILE_POINTER;
     if (r == INVALID_SET_FILE_POINTER) {
         SaveLastError();
+        return FALSE;
+    }
+    return TRUE;
+}
+
+JNIEXPORT jlong JNICALL Java_io_questdb_std_Files_copyData
+        (JNIEnv *e, jclass cls, jlong fdFrom, jlong fdTo, jlong fromOffset, jlong length) {
+
+    char buf[16*4096];
```

Comment:
nit: 64KB sounds like too much for a stack-based allocation.

Reply:
@bziobrowski found it's better than 4k on windows

Reply:
AFAIK stack size is something like 1MB on Windows unlike Linux where most distros have it set to 8MB. In JVM it's more complex than that since JVM sets its own stack size (which may be increased with the `-Xss` flag): the default on Linux, MacOS and BSD is 1MB, while on Windows it's the OS default (again - 1MB on 64-bit version).

But since we don't have any complex code, e.g. recursion, it's probably fine to allocate 16KB on the stack. Marking this one as resolved.